### PR TITLE
[feat]Support no-sla dashboard repos

### PR DESCRIPTION
**Pull request by SavvasM1**
## Summary                                                                                                                                                                                                                       
   
Surface no-SLA repos on the dashboards so they aren't silently dropped.                                                                                                                                                          
                                                                      
- **Schema**: new `V15 has_sla` column on `pr_tracking`, set at insert and never modified — durable per-PR signal that survives PR close (which nulls `sla_deadline`/`sla_remaining`).                                           
- **API**:                                                          
  - `/in-flight-prs` and `/pr-stats` now expose `hasSla` (added to `InFlightPr`, `InFlightPrResponse`, `RepoInsights`).                                                                                                          
  - `/pr-stats` applies a config-join: `hasSla` reflects **present-day** `PrTrackingProps` config, not historical DB state. This fixes repos that only have pre-V15 closures appearing as "No SLA" and handles SLA↔no-SLA        
reconfigs correctly.                                                                                                                                                                                                             
  - `/in-flight-prs` keeps the persisted per-PR `has_sla` (row-granular, no config override).                                                                                                                                    
- **PR detection**: bot messages for no-SLA PRs now state "I'll still keep an eye on this one…"; team-label list in grouped notifications is deduped with debug logging.                                                         
- **UI**: in-flight PR list and tenant-requests dashboard render an SLA / No-SLA badge driven by the new `hasSla` field.                                                                                                          
- **Tests**: unit + functional coverage for the config-join semantics (SLA-config present/removed/flipped, casing mismatches, per-PR boundary case locking `/in-flight-prs` out of the config-join), plus new invariant tests for
 `InFlightPr` / `InFlightPrResponse` / `RepoInsights`.

### CODE REVIEW

File: api/service/src/main/java/com/coreeng/supportbot/prtracking/JdbcPrTrackingRepository.java [hunk 1]
```diff
@@ -266,9 +276,10 @@ public List<RepoInsights> getInsightsByRepo(@Nullable LocalDate dateFrom, @Nulla
                     COUNT(*) FILTER (WHERE sla_deadline < COALESCE(closed_at, now())) AS breached_count,
                     percentile_cont(0.5) WITHIN GROUP (ORDER BY lifetime) AS p50,
                     percentile_cont(0.9) WITHIN GROUP (ORDER BY lifetime) AS p90,
-                    percentile_cont(0.99) WITHIN GROUP (ORDER BY lifetime) AS p99
+                    percentile_cont(0.99) WITHIN GROUP (ORDER BY lifetime) AS p99,
+                    BOOL_OR(has_sla) AS has_sla
```
Comment (SennaSemakula) [inline api/service/src/main/java/com/coreeng/supportbot/prtracking/JdbcPrTrackingRepository.java:280]: This can return null if it's empty. We should wrap it with `COALESCE` to prevent ehe endpoint returning a 500

File: api/service/src/main/java/com/coreeng/supportbot/prtracking/RepoInsights.java [hunk 2]
```diff
@@ -21,5 +22,30 @@ public record RepoInsights(
...
+        // legitimately appears on the wire for a repo reconfigured SLA → no-SLA. Other callers
+        // of this record (today there are none beyond JdbcPrTrackingRepository + TenantInsightsController;
+        // tests may construct it directly) should treat hasSla and breachedCount as independent
+        // signals, not an enforced pair.
+    }
+
+    /** Returns a copy with {@code hasSla} replaced. Prefer this over positional reconstruction. */
+    public RepoInsights withHasSla(boolean newHasSla) {
+        if (newHasSla == hasSla) {
+            return this;
```
Comment (SennaSemakula) [inline api/service/src/main/java/com/coreeng/supportbot/prtracking/RepoInsights.java:37]: Do we have tests for this? Trying to figure out what this is required

File: api/service/src/main/java/com/coreeng/supportbot/prtracking/JdbcPrTrackingRepository.java (hunk 1 above)
Comment (SavvasM1) [reply api/service/src/main/java/com/coreeng/supportbot/prtracking/JdbcPrTrackingRepository.java:280]: Hmmm, I think this cannot actually return null, because of the group by github_repo clause which should guarantee every group has >=1 row, so BOOL_OR cant return null from an empty group. -- has_sla is not null defaults to false, and we already short circuit on null at the row mapping layer.

File: api/service/src/main/java/com/coreeng/supportbot/prtracking/RepoInsights.java (hunk 2 above)
Comment (SavvasM1) [reply api/service/src/main/java/com/coreeng/supportbot/prtracking/RepoInsights.java:37]: No direct test for it. It's transitively covered by the controller tests (TenantInsightsControllerTest). It's a convenience "copy-with-one-field helper". We override hasSla at read time because                slaDeadline/slaRemaining get nulled on close, so we trust today's config instead."

**Review by SennaSemakula — APPROVED** [review]
